BuildingBlocks/src/BuildingBlocks.SDK/Endpoints/Common/EndpointClient.cs

```diff
@@ -82,9 +82,19 @@ private async Task<ApiResponse<T>> Execute<T>(HttpRequestMessage request)
             responseContent = new MemoryStream(Encoding.UTF8.GetBytes(EMPTY_RESULT));
         }
 
-        var deserializedResponseContent = JsonSerializer.Deserialize<ApiResponse<T>>(responseContent, _jsonSerializerOptions)!;
+        var deserializedResponseContent = statusCode is HttpStatusCode.OK or HttpStatusCode.Created ? JsonSerializer.Deserialize<ApiResponse<T>>(responseContent, _jsonSerializerOptions)! : new ApiResponse<T>();
```

Comment:
Hmm, this creates a valid ApiResponse even though the status code is an unsuccessful one, doesn't it?